state/src/storage/tx/order_book.rs

```diff
@@ -210,15 +210,28 @@ impl<'db> StateTxn<'db, RW> {
 
     /// Delete an order from the order book
     pub fn delete_order(&self, order_id: &OrderIdentifier) -> Result<(), StorageError> {
-        self.inner().delete(ORDERS_TABLE, &order_key(order_id)).map(|_| ())
+        let order_key = order_key(order_id);
+        let order_info = self.get_order_info(order_id)?.ok_or(order_not_found(order_id))?;
+        self.inner().delete(ORDERS_TABLE, &order_key)?;
+
+        // Remove from the nullifier set
+        self.remove_from_nullifier_set(order_info.public_share_nullifier, order_id)?;
+        if order_info.local {
+            self.remove_local_order(order_id)?;
+        }
+
+        // Remove from the priority table
+        self.inner().delete(PRIORITIES_TABLE, order_id)?;
+        Ok(())
     }
 
     /// Nullify the orders indexed by the given nullifier
     pub fn nullify_orders(&self, nullifier: Nullifier) -> Result<(), StorageError> {
         // Delete all orders
         let orders = self.get_orders_by_nullifier(nullifier)?;
         for order_id in orders {
-            self.delete_order(&order_id)?;
+            let order_key = order_key(&order_id);
+            self.inner().delete(ORDERS_TABLE, &order_key).unwrap();
```

Comment:
nit: may want to use `?` instead of unwrapping to propagate errors (all other `.delete()` do this)